Wikipedia/Code/SessionSingleton.m

```diff
@@ -201,6 +201,14 @@ - (void)fetchFinished:(id)sender
                 [self cloneSessionCookies];
             }
             break;
+            case FETCH_FINAL_STATUS_FAILED: {
+                // If autoLogin fails the credentials need to be cleared out if they're no longer valid so the
+                // user has an indication that they're no longer logged in.
+                if (error.code == LOGIN_ERROR_PASSWORD_WRONG){
```

Comment:
Regarding clearing out credentials, one by one...

`LOGIN_ERROR_NAME_REQUIRED`
Auto login isn't triggered unless there's a name.

`LOGIN_ERROR_NAME_ILLEGAL`
If the user enters an illegal name, it seems it would be easier to just require they edit the name - i.e. remove a the disallowed character etc, than be forced to completely re-enter the name.

`LOGIN_ERROR_NAME_NOT_FOUND`
If "corey" is taken if we blank it out they'd have to completely re-enter "corey2" instead of just appending "2".

`LOGIN_ERROR_PASSWORD_REQUIRED`
Not good to blank out the user name because they forgot to enter a pwd.

`LOGIN_ERROR_PASSWORD_WRONG`
Not good to blank out the user name because they entered wrong pwd.

`LOGIN_ERROR_THROTTLED`
Not good to blank out the user name because their connection was throttled.

`LOGIN_ERROR_BLOCKED`
Doesn't make any difference either way.